### ISSUE TITLE
Error: Server Execution Error: Error(s) in loading state_dict for ImageProjModel

### ISSUE THREAD

**Issue by dCo3lh0** [Acly/krita-ai-diffusion#63]
I am getting this error:

> Server Execution Error: Error(s) in loading state_dict for ImageProjModel: size mismatch for proj.weight: copying a param with shape torch.Size([8192, 1024]) from checkpoint, the shape in current model is torch.Size([8192, 768]).

**Comment by dCo3lh0** [comment]
i've just noticed that it only happens when i have selected something and press generate, so the selection works fine, krita-ai-tools works fine by itself, but when i ask from ComfyUI using krita-ai-diffusion plugin to generate the image, it gives me that error.

**Comment by Acly** [comment]
The pipeline for generating with a selection mask is more complex. It's probably not directly related to how the selection was made.

The error message sounds like something you would get when using the wrong clip-vision encoder. That's the file in `ComfyUI/models/clip_vision/SD1.5/pytorch_model.bin`. Did you use the plugin's installer, or your own Comfy install?

**Comment by juan-manuel-alberro** [comment]
I'm facing the same issue using my custom Comfy installation. 
The CLIP vision in the doc is poiting to .safetensor file and not to a .bin file - https://github.com/Acly/krita-ai-diffusion/blob/main/doc/comfy-requirements.md#shared, is that correct?

**Comment by Acly** [comment]
Ah I got a bit ahead of myself there, sorry about that. The safetensors version _is_ supported now in [version 1.7.1](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.7.1) but the documentation page was already updated before I made the release.

**Comment by Azeirah** [comment]
> Ah I got a bit ahead of myself there, sorry about that. The safetensors version _is_ supported now in [version 1.7.1](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.7.1) but the documentation page was already updated before I made the release.

Is there a way to view the version of the currently installed plugin? I installed it less than half an hour ago. 

Following the documentation on my own ComfyUI, I get the same error. I installed the .safetensors clip vision model as described in the documentation.

**Comment by Acly** [comment]
You can find a `client.log` file in a folder `ai_diffusion/.logs` where you installed the plugin. It contains the version of the plugin, and  if you have 1.7.1, also about which models it found, where it searched, and what is missing.

Adding the version number somewhere in the UI would not be a bad idea though...

**Comment by noprompt** [comment]
I am also seeing this error. I have installed pretty much everything through the manager.

Two scenarios trigger this error.

1. Generate and image with SDXL, select a portion of the canvas, press generate.

<img width="709" alt="image" src="https://github.com/Acly/krita-ai-diffusion/assets/541996/602101bb-e24c-4cea-9f72-627bbeedbc3a">

2. Switch to live generate mode with SD1.5/SDXL, use image input with a layer to generate.

<img width="688" alt="image" src="https://github.com/Acly/krita-ai-diffusion/assets/541996/948c2710-9a5b-4809-9df2-976624681d73">

Here are my logs:

```
2023-11-22 22:34:41,279 INFO Connecting to http://127.0.0.1:8188
2023-11-22 22:34:41,393 INFO Found ControlNet model inpaint for SD 1.5: control_v11p_sd15_inpaint.pth
2023-11-22 22:34:41,393 INFO Found ControlNet model scribble for SD 1.5: control_v11p_sd15_scribble.pth
2023-11-22 22:34:41,394 INFO Found ControlNet model line_art for SD 1.5: control_v11p_sd15_lineart.pth
2023-11-22 22:34:41,394 INFO Optional ControlNet model line_art for SD XL not found (search path: control-lora-sketch-rank, sai_xl_sketch_)
2023-11-22 22:34:41,394 INFO Found ControlNet model soft_edge for SD 1.5: control_v11p_sd15_softedge.pth
2023-11-22 22:34:41,394 INFO Found ControlNet model canny_edge for SD 1.5: control_v11p_sd15_canny.pth
2023-11-22 22:34:41,394 INFO Optional ControlNet model canny_edge for SD XL not found (search path: control-lora-canny-rank, sai_xl_canny_)
2023-11-22 22:34:41,395 INFO Found ControlNet model depth for SD 1.5: control_v11f1p_sd15_depth.pth
2023-11-22 22:34:41,395 INFO Optional ControlNet model depth for SD XL not found (search path: control-lora-depth-rank, sai_xl_depth_)
2023-11-22 22:34:41,395 INFO Found ControlNet model normal for SD 1.5: control_v11p_sd15_normalbae.pth
2023-11-22 22:34:41,395 INFO Found ControlNet model pose for SD 1.5: control_v11p_sd15_openpose.pth
2023-11-22 22:34:41,395 INFO Optional ControlNet model pose for SD XL not found (search path: control-lora-openposexl2-rank, thibaud_xl_openpose)
2023-11-22 22:34:41,395 INFO Found ControlNet model segmentation for SD 1.5: control_v11p_sd15_seg.pth
2023-11-22 22:34:41,396 INFO Found ControlNet model blur for SD 1.5: control_v11f1e_sd15_tile_fp16.safetensors
2023-11-22 22:34:41,396 INFO Found CLIP Vision model for All: SD1.5\pytorch_model.bin
2023-11-22 22:34:41,396 INFO Found IP-Adapter model for SD 1.5: ip-adapter_sd15.bin
2023-11-22 22:34:41,396 INFO Found IP-Adapter model for SD XL: ip-adapter_sdxl_vit-h.bin
2023-11-22 22:34:41,396 INFO Found LCM LoRA model for SD 1.5: lcm-lora-sdv1-5\lcm-lora-sdv1-5.safetensors
2023-11-22 22:34:41,397 INFO Found LCM LoRA model for SD XL: lcm\SDXL\pytorch_lora_weights.safetensors
2023-11-22 22:34:41,397 INFO SD 1.5: supported
2023-11-22 22:34:41,397 INFO SD XL: supported
2023-11-22 22:48:45,842 ERROR Job 1950c18f-ffaf-4b90-a98b-17035c184634 failed: Error(s) in loading state_dict for ImageProjModel:
	size mismatch for proj.weight: copying a param with shape torch.Size([8192, 1024]) from checkpoint, the shape in current model is torch.Size([8192, 768]).
['  File "D:\\ComfyUI_windows_portable\\ComfyUI\\execution.py", line 153, in recursive_execute\n    output_data, output_ui = get_output_data(obj, input_data_all)\n', '  File "D:\\ComfyUI_windows_portable\\ComfyUI\\execution.py", line 83, in get_output_data\n    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)\n', '  File "D:\\ComfyUI_windows_portable\\ComfyUI\\execution.py", line 76, in map_node_over_list\n    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))\n', '  File "D:\\ComfyUI_windows_portable\\ComfyUI\\custom_nodes\\ComfyUI_IPAdapter_plus\\IPAdapterPlus.py", line 390, in apply_ipadapter\n    self.ipadapter = IPAdapter(\n', '  File "D:\\ComfyUI_windows_portable\\ComfyUI\\custom_nodes\\ComfyUI_IPAdapter_plus\\IPAdapterPlus.py", line 178, in __init__\n    self.image_proj_model.load_state_dict(ipadapter_model["image_proj"])\n', '  File "D:\\ComfyUI_windows_portable\\python_embeded\\lib\\site-packages\\torch\\nn\\modules\\module.py", line 2041, in load_state_dict\n    raise RuntimeError(\'Error(s) in loading state_dict for {}:\\n\\t{}\'.format(\n']
```

**Comment by juan-manuel-alberro** [comment]
Is the same issue I had, check my previous comment and download CLIP vision from that link.

**Comment by Latentnaut** [comment]
> Is the same issue I had, check my previous comment and download CLIP vision from that link.

It works. Thank you!

**Comment by Zamol-prototype** [comment]
> I'm facing the same issue using my custom Comfy installation. The CLIP vision in the doc is poiting to .safetensor file and not to a .bin file - https://github.com/Acly/krita-ai-diffusion/blob/main/doc/comfy-requirements.md#shared, is that correct?
> 
> **edit**: the issue is solved by downloading the proper .bin file from https://huggingface.co/h94/IP-Adapter/blob/main/models/image_encoder/pytorch_model.bin 😄

Is this the final solution can fix this issue ?

**Comment by Acly** [comment]
> Is this the final solution can fix this issue ?

No, the linked .safetensors file shold work (with the corrent name & folder)
